docs/generate-time-series/go.rst

```diff
@@ -0,0 +1,374 @@
+`standard library`_ that will be used in this tutorial and declare a main 
+function which will be the entry point of our executable program:
+
+.. code-block:: js
+
+    package main
+
+    import (
+        "encoding/json"
+        "fmt"
+        "io/ioutil"
+        "log"
+        "net/http"
+    )
+
+    func main() {
+
+    }
+
+Then, look at the JSON data that gets returned from going to the Open
+Notify API endpoint at `<http://api.open-notify.org/iss-now.json>`_. The 
+endpoint returns a JSON payload, which contains an ``iss_position`` object
+with ``latitude`` and ``longitude`` data.
+
+.. code-block:: js
+
+   {
+        "message":"success",
+        "timestamp":1591703638,
+        "iss_position":{
+            "longitude":"84.9504",
+            "latitude":"41.6582"
+        }
+    }
+
+The longitude and latitude of the International Space Station changes 
+constantly and is what you want to extract from this payload and insert into
+CrateDB. To parse this JSON, you can create a `struct`_ that can be used to 
+`unmarshal`_ the data into. When you unmarshal JSON into a struct, the 
+function matches incoming object keys to the keys in the struct field name
+or its tag. By default, object keys which don't have a corresponding struct 
+field are ignored.
+
+.. code-block:: js
+
+    type issInfo struct {
+        IssPosition struct {
+            Longitude string `json:"longitude"`
+            Latitude  string `json:"latitude"`
+        } `json:"iss_position"`
+    }
+
+Now, create a function that makes an HTTP GET request to the Open Notify API 
+endpoint and returns longitude and latitude as a `geo_point`_ declaration.
+
+The Go standard library comes with great support for HTTP client and server
+implementations in the `net/http`_ package. In this example you’ll use it to 
+issue an HTTP GET request to the API endpoint. 
+
+Then you will read the response body and unmarshal the JSON into your defined 
+struct ``issInfo``. 
+
+Lastly, you will format the return string and return it. 
+
+.. code-block:: js
+
+    func getISSPosition() (string, error) {
+        var i issInfo
+
+        response, err := http.Get("http://api.open-notify.org/iss-now.json")
+        if err != nil {
+            return "", fmt.Errorf("unable to retrieve request: %v", err)
+        }
+        defer response.Body.Close()
+
+        if response.StatusCode/100 != 2 {
+            return "", fmt.Errorf("bad response status: %s", response.Status)
+        }
+
+        responseData, err := ioutil.ReadAll(response.Body)
+        if err != nil {
+            return "", fmt.Errorf("unable to read response body: %v", err)
+        }
+
+        err = json.Unmarshal(responseData, &i)
+        if err != nil {
+            return "", fmt.Errorf("unable to unmarshal response body: %v", err)
+        }
+
+        s := fmt.Sprintf("(%s, %s)", i.IssPosition.Longitude, i.IssPosition.Latitude)
+        return s, nil
+    }
```

Comment:
I would introduce this code block with something like, "Here's a function that [...]:"

then afterward, I would consider adding some text to explain what is happening, step by step. e.g., "Above, the `getISSPosition` function does the following:" -- you might want to take the explanation above this function and move it here to serve this purpose

then, you can list out a bunch things, explaining the function as you go

---

to help illustrate my point, here's an excerpt from the tutorial I am working on at the moment that appears directly after a Python code block:

> Above, the ``plot()`` function:
>
> * Generates a [figure]() that measures 12 × 6 (inches)
> * Plots ``data`` as a [scatter]() diagram (distance over time)
> * Sets the [axes]() labels and title
> * Sets up the x-axis to [handle datetimes]()
> * Configures major [tick locations]() every [hour]()
> * Configures major [tick formatting]() with a [time string]() (``%H:00``)
> * Forces Matplotlib to plot the whole data set, including null ``time`` values, by manually setting the [limits of the x-axis]() (which are trimmed by default)
> * Activates x-axis tick label [auto-formatting]() (rotates them for improved readability)